runtime/common/tests/xcm_tests.rs

```diff
@@ -2156,3 +2157,579 @@ fn less_than_min_xcm_fee_should_not_work() {
+        Some(units_per_sec),
+        None,
+    );
+    // Register relaychain native asset in ParaB
+    let relay_asset_id_on_b = register_assets_on_parachain::<ParaB>(
+        &relay_source_location,
+        &relay_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+
+    let relay_asset_amount_minted_on_a = 10000000;
+    // Initialize some relay chain tokens for alice
+    assert_ok!(ParaA::execute_with(|| {
+        parachain::Assets::mint(
+            parachain::Origin::signed(parachain::AssetManager::account_id()),
+            relay_asset_id_on_a,
+            ALICE,
+            relay_asset_amount_minted_on_a,
+        )
+    }));
+
+    let dest = MultiLocation {
+        parents: 1,
+        interior: X2(
+            Parachain(para_a_id),
+            AccountId32 {
+                network: NetworkId::Any,
+                id: ALICE.into(),
+            },
+        ),
+    };
+
+    // Send some ParaB tokens from Alice on B to Alice on A
+    let amount_to_a = 10000000;
+    let weight = 40;
+    ParaB::execute_with(|| {
+        assert_ok!(parachain::XTokens::transfer(
+            Some(ALICE).into(),
+            parachain::CurrencyId::MantaCurrency(b_asset_id_on_b),
+            amount_to_a,
+            Box::new(VersionedMultiLocation::V1(dest.clone())),
+            weight,
+        ));
+    });
+
+    let dest = MultiLocation {
+        parents: 1,
+        interior: X2(
+            Parachain(para_b_id),
+            AccountId32 {
+                network: NetworkId::Any,
+                id: ALICE.into(),
+            },
+        ),
+    };
+
+    let amount_back_to_b = 100;
+    let fee_amount: u128 = 50;
+    let min_xcm_fee = 10;
+    // Send some ParaB tokens from Alice on A back to Alice on B
+    ParaA::execute_with(|| {
+        assert_ok!(AssetManager::set_min_xcm_fee(
+            parachain::Origin::root(),
+            para_b_source_location,
+            min_xcm_fee,
+        ));
+
+        assert_ok!(parachain::XTokens::transfer_multicurrencies(
+            Some(ALICE).into(),
+            vec![
+                (
+                    parachain::CurrencyId::MantaCurrency(b_asset_id_on_a),
+                    amount_back_to_b
+                ),
+                (
+                    parachain::CurrencyId::MantaCurrency(relay_asset_id_on_a),
+                    fee_amount
+                )
+            ],
+            1,
+            Box::new(VersionedMultiLocation::V1(dest.clone())),
+            weight,
+        ));
+
+        assert_eq!(
+            parachain::Assets::balance(relay_asset_id_on_a, &ALICE),
+            relay_asset_amount_minted_on_a - fee_amount
+        );
+        assert_eq!(
+            // notice that total supply between the two chains is:
+            // (relay_asset_amount_minted_on_a - fee_amount) + (fee_amount - min_xcm_fee)
+            parachain::Assets::total_supply(relay_asset_id_on_a),
+            relay_asset_amount_minted_on_a - fee_amount
```

Comment:
according to your comment above this should be `relay_asset_amount_minted_on_a - min_xcm_fee`

Which one is correct?

Reply:
The comment is not great. The total supply should be counted from both chains:
`relay_asset_amount_minted_on_a - fee_amount` is still on `ParaA`
`fee_amount - min_xcm_fee` is on ParaB.
Will rewrite it.

Reply:
So then the comment is correct and the logic is 2311 isn't, right?

Reply:
Hmm something weird happened here, I wrote a big comment but it wasn't pushed.

Reply:
https://github.com/Manta-Network/Manta/pull/576/commits/a140e5a016635b48e2abcf985d84d2e98461b4fc This is the commit with the change, I don't know why it's not reflected in this diff.
Also I remember resolving this conversation, but somehow it is not resolved...

Reply:
Actually none of my changes from the last 3 commits are reflected

Reply:
Please refresh the diff, should be showing up correctly now , but I think I understood your question about 2311:
```
                               (
					parachain::CurrencyId::MantaCurrency(relay_asset_id_on_a),
					fee_amount
				)
```
This is the fee being sent to ParaB, but the logic in ORML will subtract `min_xcm_fee` so the comment is `fee_amount - min_xcm_fee is on ParaB.` is correct. Let me write this explicitly.

Reply:
ok, got it thanks. i thought total_supply was calculated over both chains, not from the view of ParaA

Also, I unresolved this comment earlier to make sure you see it before merging